src/js/ParticleSystem.js

```diff
@@ -0,0 +1,123 @@
+import THREE from 'three';
+
+function applyPointRandomness(r) {
+  let p = new THREE.Vector3(
+    (Math.random() - 0.5),
+    (Math.random() - 0.5),
+    (Math.random() - 0.5))
+    .normalize()
+    .multiplyScalar(r);
+  return p;
+}
+
+function newParticlePosition(oldPosition, newPosition, lerpFactor, offset, pointRandomness) {
+  let result = new THREE.Vector3();
+  result.copy(oldPosition);
+  result.lerp(newPosition.clone().add(offset), lerpFactor);
+  result.add(applyPointRandomness(pointRandomness));
+  return result;
+}
+
+class ParticleSystem {
+
+  constructor(scene) {
+    this.emitters = [];
+    this.scene = scene;
+  }
+
+  update(delta) {
+    this.emitters.forEach(emit => {
+      this.updateEmitter(emit, delta);
+    });
+  }
+
+  createEmitter(options) {
+    let emitter = new THREE.Points();
+
+    emitter.offset = options.offset;
+    emitter.oldPosition = emitter.offset.clone();
+    emitter.bindTo = options.bindTo;
+    emitter.spawnRate = options.spawnRate;
+    emitter.lifetime = options.lifetime;
+    emitter.velocity = options.velocity;
+    emitter.velocityRandomness = options.velocityRandomness;
+    emitter.pointRandomness = options.pointRandomness;
+
+    emitter.geometry = new THREE.Geometry();
+    let toSpawn = Math.ceil(emitter.spawnRate * emitter.lifetime);
+    for (let i = 0; i < toSpawn; i++) {
+      let point = newParticlePosition(
+        emitter.oldPosition,
+        emitter.bindTo.position,
+        1,
+        emitter.offset.clone().applyQuaternion(emitter.bindTo.quaternion),
+        emitter.pointRandomness
+      );
+      emitter.geometry.vertices[i] = point;
+      point.velocity = emitter.velocity.clone();
+      point.velocity.multiplyScalar(1 - emitter.velocityRandomness * (2 * Math.random() - 1));
```

Comment:
Couldn't the velocity also be part of the `newParticlePosition` (or `newParticle`) function?